geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/util/GeoHashUtils.scala

```diff
@@ -0,0 +1,165 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.spark.jts.util
+
+import com.vividsolutions.jts.geom.{Envelope, Geometry, GeometryFactory, PrecisionModel}
+import scala.collection.BitSet
+import scala.collection.immutable.{BitSet => IBitSet}
+
+object GeoHashUtils {
+
+  def encode(geom: Geometry, precision: Int): String = {
+    GeoHash.encode(geom.getCentroid.getX, geom.getCentroid.getY, precision)
+  }
+
+  def decode(geohash: String, precision: Int): Geometry = {
+    GeoHash.decode(geohash, Some(precision))
+  }
+}
+
+case class Bounds(low: Double,
+                  high: Double) {
+  lazy val mid = (low+high)/2.0
+}
+
+object GeoHash {
```

Comment:
did you code this up from scratch? if not, we should add a comment indicating where it came from, and might need to include copyright headers

Reply:
the GeoHash object is just pickings from from utils.geohash.GeoHashUtils.
the GeoHashUtils object is from scratch. I'll make a note, but I'm not sure what headers to need.

Reply:
if it's copied from inside geomesa, then just make sure that you copy any non-ccri copyrights from the class you copied from. i wasn't sure where it was from, or if you had coded it up, since i didn't see some of the methods in geomesa-utils, like `powersOf2Map`.

Reply:
but figured you probably didn't re-implement it given that there are a bunch of implementations already out there

Reply:
Got it. I also borrowed some bits from utils.geohash.GeoHash but it's all from inside geomesa. I'll look through the comments in those two files